libsolidity/ast/Types.cpp

```diff
@@ -1864,14 +1864,15 @@ TypePointer ArrayType::decodingType() const
 TypePointer ArrayType::interfaceType(bool _inLibrary) const
 {
 	// Note: This has to fulfill canBeUsedExternally(_inLibrary) ==  !!interfaceType(_inLibrary)
```

Comment:
I just created #6223, because this is in fact horrible IMO...

Reply:
I agree. I was just too much focused on "How the hell is this supposed to work" to have any capacity left for "How could this work instead" ;)

Reply:
Yeah - fixing this is a job for a separate PR anyways :-).